src/components/authors/manageAuthorPage.js

```diff
@@ -74,20 +66,26 @@ var ManageAuthorPage = React.createClass({
 			AuthorActions.createAuthor(this.state.author);
 		}
 		
-		this.setState({dirty: false});
-		toastr.success('Author saved.');
-		this.transitionTo('authors');
+		this.setState({dirty: false, redirect: true}, function() {
+			toastr.success('Author saved.');
+		});
 	},
 
 	render: function() {
 		return (
-			<AuthorForm
-				author={this.state.author}
-				onChange={this.setAuthorState}
-				onSave={this.saveAuthor}
-				errors={this.state.errors} />
+			<div>
+				{ this.state.redirect && <Redirect to="/authors" />}
+				<Prompt when={this.state.dirty} message="Leave without saving?" />
+				<AuthorForm

```

Comment:
- Use React Router 4's `Redirect` component instead of `transitionTo`.
- Use React Router 4's `Prompt` component instead of `willTransitionFrom`.